### ISSUE TITLE
Not all features work on Pi.

### ISSUE THREAD

**Issue by ItsTerm1n4l** [WayfireWM/wayfire#1966]
**Describe the bug**
There are various features that I have not been able to get to work on PI OS bookworm, namely;
when I try to set cubemap mode for the `cube` plugin I just see a green background instead of my image, it's the same for the skydome mode.
**To Reproduce**
1. Enable plugin cube.
2. go in cube view
3. I see a green background.

**Expected behavior**
I should see my image as the background in cube view.

**My config**
```
[command]
#binding_kill = <super> <shift> KEY_Q
repeatable_binding_volume_up=KEY_VOLUMEUP
command_volume_up=wfpanelctl volumepulse volu
repeatable_binding_volume_down=KEY_VOLUMEDOWN
command_volume_down=wfpanelctl volumepulse vold
binding_mute=KEY_MUTE
command_mute=wfpanelctl volumepulse mute
binding_menu=<super>
command_menu=wfpanelctl smenu menu
binding_terminal=<ctrl> KEY_ENTER
command_terminal=alacritty
binding_terminalsize=<ctrl> <shift> KEY_ENTER
command_terminalsize=slurp -f %w,%h | awk -F',' '{ printf "-o window.dimensions.columns=%d -o window.dimensions.lines=%d", $1/8,$2/18 }' | xargs alacritty
binding_bluetooth=<ctrl> <alt> KEY_B
command_bluetooth=wfpanelctl bluetooth menu
binding_netman=<ctrl> <alt> KEY_W
command_netman=wfpanelctl netman menu
binding_grim=KEY_SYSRQ
command_grim=grim
binding_orca=<ctrl> <alt> KEY_SPACE
command_orca=gui-pkinst orca reboot
binding_quit=<ctrl> <alt> KEY_DELETE
command_quit=lxde-pi-shutdown-helper
binding_power=KEY_POWER
command_power=pwrkey

[input-device:generic ft5x06 (79)]
output=DSI-1

[input-device:generic ft5x06 (80)]
output=DSI-1

[input-device:FT5406 memory based driver]
output=DSI-1

[input]
xkb_model=pc105
xkb_layout=gb
xkb_variant=
kb_repeat_delay=400
kb_repeat_rate=40
mouse_cursor_speed=0
left_handed_mode=false

[core]
background_color = \#1A1A1AFF
close_top_view = <ctrl> <shift> KEY_Q
preferred_decoration_mode=server
# List of plugins to be enabled.
# See the Configuration document for a complete list.
plugins = \
  animate \
  cube \
command \
 decoration \
  wobbly \
  move \
resize \
switcher \
scale \
alpha \
autostart \
window-rules \
#blur \

[autostart]
waypaper=python3 /home/pi/Documents/waypaper-2.0.2-py3.11/waypaper/__main__.py --restore
eww=eww open bar

[move]
activate = <ctrl> BTN_LEFT
enable_snap=true
workspace_switch_after = 1

[resize]
activate = <super> BTN_RIGHT

[cube]
activate=<super> KEY_T
background_mode=image
deform=2
zoom=0.5
cubemap_image=/home/pi/.wallpapers/original.png
background = \#151724FF

[decoration]
active_color = \#EBCE8BFF
border_size = 4
button_order = minimize maximize close
font = monospace
ignore_views = none
inactive_color = \#A3BE8CFF   #OLD was 181926FF
title_height = 0

#[blur]
#method=gaussian
#kawase_degrade = 1
#kawase_iterations = 2
#kawase_offset = 5.000000
#mode=normal 
#toggle=none

[switcher]
#gesture_toggle = <super> KEY_T
speed=200

[window-rules]
alacritty = on created if app_id is "Alacritty" then set alpha 0.8
chromium-browser = on created if app_id is "Chromium-browser" then maximize

[scale]
toggle=<ctrl> KEY_P 
toggle_all
interact=true
middle_click_close
inactive_alpha=0.5
```
**Wayfire version**
0.7.5
Thank you.

**Comment by ammen99** [comment]
I see that you have not changed the background_mode option for `cube` ;) You should set it to `cubemap` or `skydome`, respectively.

Also, does the Pi support GLES 3.2 or tesselation shaders at least? Those are required for the `deform` option.

**Comment by ItsTerm1n4l** [comment]
Sorry, I had edited background_mode to `cubemap` but had changed it back to `simple` so I didn't have a glaring green screen, then when I uploaded my config I forgot and set `image` instead of `cubemap`, but with background_mode set to `cubemap` it is still green.
I don't know what version of GLES the Pi supports, I looked online but was unable to find it, I did however try setting `deform` to `1` and then `2` but it didn't make any difference so I don't think it does support GLES 3.2.

**Comment by ammen99** [comment]
> I don't know what version of GLES the Pi supports, I looked online but was unable to find it, I did however try setting `deform` to `1` and then `2` but it didn't make any difference so I don't think it does support GLES 3.2.

Are there any errors in the log (log is printed on stdout)? Also, if you don't have GLES 3.2 you should use `deform=0` which is the default cube without anything fancier.

**Comment by ItsTerm1n4l** [comment]
I tried `cat /dev/stdout` and nothing came up, I know about `deform=0` I just had forgotten to set it back to that lol.

**Comment by ammen99** [comment]
> cat /dev/stdou

I don't think this will work. How do you start Wayfire?

**Comment by ItsTerm1n4l** [comment]
I dont know exactly, it is a new install of RPI OS and had wayfire pre-installed.

**Comment by ItsTerm1n4l** [comment]
There is a file at `/usr/share/wayland-sessions/wayfire.desktop` which contains
```
[Desktop Entry]
Name=Wayfire
Exec=wayfire
TryExec=wayfire
Icon=
Type=Application
DesktopNames=Wayfire
```

**Comment by spl237** [comment]
cube did certainly work on the Pi at one point, but bear in mind we are in the process of doing some changes to the graphics backend which may cause some features, particularly those involving 3D, to be broken or glitchy.

wayfire itself is started by the wayfire-pi script in /usr/bin - this sets up a bunch of environment variables which are needed to get it working correctly with our desktop environment.

**Comment by PeterOGB** [comment]
Unless you've changed something else, the file used will be /usr/share/wayland-sessions/LXDE-pi-wayfire.desktop
I think the wayfire.desktop is left over from before they started their own customised desktop.  It should probably be be removed.

**Comment by ItsTerm1n4l** [comment]
`pgrep wayfire` gives a PID though whereas `pgrep wayfire-pi` does not.

**Comment by spl237** [comment]
Yes, that's because wayfire-pi is a script wrapping wayfire; wayfire is the binary executable, so it is what is shown in ps. But it will be wayfire-pi which is called in the LXDE-pi-wayfire session mentioned above.

**Comment by ItsTerm1n4l** [comment]
Ah ok, got it.

**Comment by ItsTerm1n4l** [comment]
I have fixed the issue, I have finally been able to install WCM thanks to https://github.com/WayfireWM/wayfire/issues/1965 and now have been able to work it out. the problem was with the `skydome` option I had to specify an image, I had not seen this in the documentation. And for the `cubemap` option I had to use a special cubemap image, not just any wallpaper.
Maybe the docs should be updated with more information to make this more clear.
Thank you.
